### ISSUE TITLE
Really cool work

### ISSUE THREAD

**Issue by LysandreJik** [remyxai/FFMPerative#1]
Really cool work, loved playing around with it!

If ever you're interested in having other users re-use your tools, you can also push them to the Hub ([see the `push_to_hub` call in this guide](https://huggingface.co/docs/transformers/custom_tools#creating-a-new-tool)), but it's, of course, not required :)

Is there anything we can do from `transformers` side to help you on this cool project? Did you run into any issues we can help fix?

**Comment by smellslikeml** [comment]
@LysandreJik Thanks again for trying it out! We think the hub is a great way to increase accessibility.

We found it incredibly easy to get the agent interface working, and building custom tools was a breeze. A few things we observed while we were developing:

* The `push_to_hub()` method errors out when pushing a tool with external dependencies. It would be great to have the ability to set a requirements.txt file when calling this method. 
* Can a collection of `Tool`s be uploaded to a single space? Since they all have common external dependencies, it may make sense to bundle them together.

Kudos to you and your team for creating an excellent framework for agents. 👏

**Comment by sgugger** [comment]
> Can a collection of Tools be uploaded to a single space? 

Our thinking was that each space would be a demo of the Tool, which doesn't work with multiple tools in the same space. There is however a function to get all tools repo from an organization if that helps.

**Comment by smellslikeml** [comment]
Makes sense and that is indeed helpful! We'll try that and update our README, thanks @sgugger
